src/libraries/System.Private.CoreLib/src/System/Globalization/InvariantModeCasing.cs

```diff
@@ -0,0 +1,779 @@
+                    {
+                        // Found match.
+                        return (int) (pCurrentSource - pSource);
+                    }
+
+                    pCurrentSource++;
+                }
+
+                return -1;
+            }
+        }
+
+        internal static unsafe int LastIndexOfIgnoreCase(ReadOnlySpan<char> source, ReadOnlySpan<char> value)
+        {
+            Debug.Assert(value.Length > 0);
+            Debug.Assert(value.Length <= source.Length);
+            Debug.Assert(GlobalizationMode.Invariant);
+
+            fixed (char* pSource = &MemoryMarshal.GetReference(source))
+            fixed (char* pValue  = &MemoryMarshal.GetReference(value))
+            {
+                char* pValueLimit = pValue + value.Length - 1;
+                char* pCurrentSource = pSource + (source.Length - value.Length);
+
+                while (pCurrentSource >= pSource)
+                {
+                    char *pVal = pValue;
+                    char *pSrc = pCurrentSource;
+
+                    while (pVal <= pValueLimit)
+                    {
+                        if (!char.IsHighSurrogate(*pVal) || pVal == pValueLimit)
+                        {
+                            if (*pVal != *pSrc && ToUpper(*pVal) != ToUpper(*pSrc))
+                                break; // no match
+
+                            pVal++;
+                            pSrc++;
+                            continue;
+                        }
+
+                        if (char.IsHighSurrogate(*pSrc) && char.IsLowSurrogate(*(pSrc + 1)) && char.IsLowSurrogate(*(pVal + 1)))
+                        {
+                            // Well formed surrogates
+                            // both the source and the Value have well-formed surrogates.
+                            if (!SurrogateCasing.Equal(*pSrc, *(pSrc + 1), *pVal, *(pVal + 1)))
+                                break; // no match
+
+                            pSrc += 2;
+                            pVal += 2;
+                            continue;
+                        }
+
+                        if (*pVal != *pSrc)
+                            break; // no match
+
+                        pSrc++;
+                        pVal++;
+                    }
+
+                    if (pVal > pValueLimit)
+                    {
+                        // Found match.
+                        return (int) (pCurrentSource - pSource);
+                    }
+
+                    pCurrentSource--;
+                }
+
+                return -1;
+            }
+        }
+
+        // The following case mapping data is generated using the tool stored in in the path .\src\coreclr\System.Private.CoreLib\Tools\InvariantCasing.
+        // We generate the casing tables as 8-4-4 tables to balance between the size of the data and the performance of teh mapping operation.
+        // The data generated here is matching the Unicode version 14.
```

Comment:
We probably shouldn't use Version 14 since it hasn't been released yet. The latest released version is 13.0.0.

Also, any reason we didn't use the existing tool (https://github.com/dotnet/runtime/tree/main/src/coreclr/System.Private.CoreLib/Tools/GenUnicodeProp) to generate this data?

Reply:
I can look later at using GenUnicodeProp as I thought this more about generating the Unicode categories and not generating the real case mapping. I'll track looking at that later if you think this is not a blocker to this PR.

Reply:
Not a blocker at all. But there are two benefits: (a) it ensures all of our data is built from the same files; and (b) I believe we previously measured the size increase at much less than 12KB.

Reply:
Now I am using GenUnicodeProp and it gave a better results. thanks for pointing at that.